### PR TITLE
Apim 1409 add delete groups

### PR DESCRIPTION
**Pull request by jourdiw**
## Issue

https://gravitee.atlassian.net/browse/APIM-1409

## Description

- create GET /groups endpoint in rest-api-v2
- integrate groups page for V4 APIs
- change V1 APIs as read-only

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-dbeeiocmlg.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 1]
```diff
@@ -3791,6 +3811,67 @@ components:
...
+                id:
+                    type: string
+                    description: Group's uuid.
+                    example: 00f8c9e7-78fc-4907-b8c9-e778fc790750
+                name:
+                    type: string
+                    description: Group's name.
+                    example: Gravitee group
+                    minLength: 1
+        Group:
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:3826]: @phiz71 Any thoughts/feedback on Group in the open-api?

Comment (jourdiw) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:3826]: Applied changes as discussed 👍

**Review by loriepisicchio — APPROVED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 2]
```diff
@@ -3791,6 +3811,74 @@ components:
...
+                        example: 2018-11-23T22:00:00.000Z
+                    maxInvitation:
+                        type: integer
+                        description: Maximum number of group members.
+                    apiRole:
+                        type: string
+                        description: The Api role of the group.
+                    lockApiRole:
+                        type: boolean
+                        description: Api role cannot be changed.
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:3857]:
```suggestion
                        description: Api role cannot be changed by a group admin.
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 3]
```diff
@@ -3791,6 +3811,74 @@ components:
...
+                    applicationRole:
+                        type: string
+                        description: The Application role of the group.
+                    lockApplicationRole:
+                        type: boolean
+                        description: Application role cannot be changed.
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:3863]:
```suggestion
                        description: Application role cannot be changed by a group admin.
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 4]
```diff
@@ -3791,6 +3811,74 @@ components:
...
+                        description: Notifications of new members are disabled.
+                    primaryOwner:
+                        type: boolean
+                        description: Group's members contain one primary owner.
+        GroupEvent:
+            type: string
+            description: Type of group event.
+            enum:
+                - API_CREATE
+                - APPLICATION_CREATE
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:3881]: 👍🏻

**Review by phiz71 — APPROVED** [review]
Good job 👍🏻 .
I've made a suggestion to add details in 2 descriptions, in the OpenAPI descriptor. Up to you to change or not :)

**Review by phiz71 — APPROVED** [review]
Good job 👍🏻 .
I've made a suggestion to add details in 2 descriptions, in the OpenAPI descriptor. Up to you to change or not :)

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml (hunk 2 above)
Comment (jourdiw) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:3857]: Can it be changed by someone other than a group admin?

Comment (jourdiw) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:3857]: Just tried to follow the rabbit down the rabbit hole... It doesn't seem very obvious that only admins can change the role or that other users can change the role who are not admin. I'm just going to keep the descriptions as is for now.

**Pull request merged**